### PR TITLE
Pokemon follower changes for issue #5135

### PR DESCRIPTION
**Pull request by Liamjd14**
Will fix the following issues from #5135. (makrked the currently done ones)

Raichu-Alola follower shiny a bit too sunburnt and ears/tail too light -done
Sandshrew-Alola follower ear anim south - done
Vulpix-Alola follower east/west blinking pixels - done
Ninetales-Alola follower south blinking pixels - done
Persian-Alola follower east/west could have a less pointy face - done
Graveler-Alola follower blinking pixels - done
Golem-Alola follower blinking pixels - done
Exeggutor-Alola follower stray pixels on north sprite - done 
Marowak-Alola follower fire color, shiny palette slightly off - done
Ponyta-Galar follower shiny hair too yellow - done
Rapidash-Galar follower shiny hair too yellow - done
Growlithe-Hisui follower blinking pixels south sprite - done
Arcanine-Hisui follower blinking pixels - done
Typhlosion-Hisui follower shiny stomach color wrong - done
Sliggo-Hisui follower normal palette spots wrong color - done
Goodra-Hisui follower throat goo missing - done
Dialga-origin follower shiny missing - done (based of shiny battler)
Palkia-origin follower shiny missing - Sprite and pallets reworked to allow for shiny, might need a second pass
Shaymin-sky follower sprite missing - done (custom sprite based off of hgss sprite)
Basculin-white follower shiny missing - done
Sawsbuck-Winter follower blinking pixels, horns have transparent pixels - Done
Tornadus-Therian follower sprite missing - done
Thundurus-Therian follower sprite missing - done
Landorus-Therian follower sprite missing - done 
Enamorus-Therian follower shiny missing - done along with normal form
Kyurem-White follower sprite missing - done (see below for credits)
Kyurem-Black follower sprite missing - done (see below for credits)
Flabebe-all follower shiny flower stem - done
Floette-Eternal follower shiny wrong palette - done
Pumpkaboo/Gourgeist sizes missing sprites - Done
Zydog follower missing sprite - done (see below for credits)
Oricorio-pompom/pa'u/sensu follower missing sprite - done (see below for credits)
Necrozma-Dawn Wings follower blinking pixel south sprite above right eye - done
Toxtricity-LK follower missing shiny - done
Calyrex-Rider follower missing shiny (gold cape/crown) - done
Raticate-Totem follower missing sprite - done
Mimikyu-Totem follower missing sprite - done
Marowak-Totem follower missing sprite - done
Partner Pikachu follower missing sprite - done
Partner Eevee follower missing sprite - done
 Toxtricity-LK-Gmax has a follower sprite? Why? - done

Also Adds East/West Sprites for: Done (sprites have the frames, just need the code to make them work and a test after.
Slowbro-Galar follower should look different east and west (needs new anim code)
Tinkaton Line
Scovillain

Fix issue with Lycanroc Midday: #5319

Added: 
Original Magearna OW Sprite

##Images
pokeemerald-0
pokeemerald-1
pokeemerald-2
pokeemerald-3

##Sources and Credits
Modifying and Formatting Sprites: Liamjd14

Oricorio/ Zydog / Zycomplete/ Original Megaerna Sprites: Princess-Phoenix, Larryturbo, Kidkatt, Zender1752, SageDeoxys
Source: https://www.deviantart.com/larryturbo/art/Gen-7-Alola-Overworld-Sprites-805455576

Kurem White and Black: Larryturbo
Sources: 
https://www.deviantart.com/larryturbo/art/646-Kyurem-White-810269180
https://www.deviantart.com/larryturbo/art/646-Kyurem-Black-810385248

## Issue(s) that this PR fixes
Fixes the issues listed above from #5135 and #5319

## **Discord contact info**
Liamjd14

### CODE REVIEW

File: include/config/overworld.h [hunk 1]
```diff
@@ -43,7 +43,7 @@
                                                   // (You should not use 48x48 sprites/tables for compressed gfx)
                                                   // 16x32, 32x32, 64x64 etc are fine
 // Follower Pokémon
-#define OW_FOLLOWERS_ENABLED           FALSE      // Enables follower Pokémon, HGSS style. Requires OW_POKEMON_OBJECT_EVENTS. Note that additional scripting may be required for them to be fully supported!
+#define OW_FOLLOWERS_ENABLED           TRUE       // Enables follower Pokémon, HGSS style. Requires OW_POKEMON_OBJECT_EVENTS. Note that additional scripting may be required for them to be fully supported!
```
Comment (hedara90) [inline include/config/overworld.h:46]:
```suggestion
#define OW_FOLLOWERS_ENABLED           FALSE     // Enables follower Pokémon, HGSS style. Requires OW_POKEMON_OBJECT_EVENTS. Note that additional scripting may be required for them to be fully supported!
```

File: graphics/pokemon/magearna/original_color/overworld.png
Comment (hedara90) [inline graphics/pokemon/magearna/original_color/overworld.png:1]: Massive blinking on north and south.
Shiny very wrong colors.

File: graphics/pokemon/calyrex/shadow_rider/overworld_shiny.pal
Comment (hedara90) [inline graphics/pokemon/calyrex/shadow_rider/overworld_shiny.pal:1]: Blinking left side of crown on south-

File: graphics/pokemon/calyrex/ice_rider/overworld_shiny.pal
Comment (hedara90) [inline graphics/pokemon/calyrex/ice_rider/overworld_shiny.pal:1]: Missing top of thingies on sides of head, should match Shadow rider-

**Review by hedara90 — CHANGES_REQUESTED** [review]
Some fixes to do.

**Comment by hedara90** [comment]
As a temporary thing I'd merge in the Pumkaboo and Totem followers. But ideally those should have different sprites for different sizes.

File: graphics/pokemon/sliggoo/hisuian/overworld.png
[the remark this answers is not in the document]
Comment (Liamjd14) [reply graphics/pokemon/sliggoo/hisuian/overworld.png:1]: modifled to be exact match to shiny battle sprite

**Comment by Liamjd14** [comment]
I plan on adding one other change to this if a bit. Should fix the Size problems with the Pumpkaboo line and totums

**Comment by Liamjd14** [comment]
I dont know if its worth noting but all the totems use the regular pokemons ow sprites with no scaling as far as i can tell.

**Comment by AsparagusEduardo** [comment]
> I dont know if its worth noting but all the totems (even the ones i didnt edit) use the regular pokemons ow sprites with no scaling as far as i can tell.

I think it is worth noting, thank you :)

**Review by hedara90 — CHANGES_REQUESTED** [review]
Just 2 things

**Comment by Liamjd14** [comment]
pokeemerald-0
pokeemerald-1
 Both are fixed

**Review by hedara90 — APPROVED** [review]

**Pull request merged**